File: Chapter5/R/Example1.R
# Chapter 5 - Other Discrete Distributions
# Example 1: Geometric Distribution  p = 0.3

# Note: R's dgeom/pgeom counts FAILURES before first success.
# To work with TRIALS (X = failures + 1), subtract 1 from the argument.

# Parameter
p <- 0.3

# P(X = 5)  →  5th trial is first success = 4 failures
prob_exact <- dgeom(4, p)
cat(sprintf("P(X = 5) = %.4f\n", prob_exact))

# P(X <= 4)  →  at most 4 trials = at most 3 failures
prob_cumulative <- pgeom(3, p)
cat(sprintf("P(X <= 4) = %.4f\n", prob_cumulative))

# P(X > 6)  →  more than 6 trials = more than 5 failures
prob_greater <- 1 - pgeom(5, p)
cat(sprintf("P(X > 6) = %.4f\n", prob_greater))

# Expected number of trials
expected_trials <- 1 / p
cat(sprintf("Expected trials: %.2f\n", expected_trials))

File: Chapter5/Python/Example1.py
# Chapter 5 - Other Discrete Distributions
# Example 1: Geometric Distribution  p = 0.3

# Note: scipy's geom counts TRIALS to first success, starting from k=1.

from scipy.stats import geom

# Parameter
p = 0.3

# P(X = 5)
prob_exact = geom.pmf(5, p)
print(f"P(X = 5) = {prob_exact:.4f}")

# P(X <= 4)
prob_cumulative = geom.cdf(4, p)
print(f"P(X <= 4) = {prob_cumulative:.4f}")

# P(X > 6) = 1 - P(X <= 6)
prob_greater = 1 - geom.cdf(6, p)
print(f"P(X > 6) = {prob_greater:.4f}")

# Expected number of trials
expected_trials = 1 / p
print(f"Expected trials: {expected_trials:.2f}")
